### DISCUSSION TITLE
Passing variables for use in config.yml?

### DISCUSSION THREAD

**Discussion by notflip** [borgmatic-collective/docker-borgmatic#288] [Q&A] [answered]
What's the best way to create a general script, based on the example from this repo, which allows us to pass variables to the config.yaml file inside /data/borgmatic.d/config.yaml?

Thanks!

**Comment by witten** [comment, marked as answer]
How are the variables intended to be used in borgmatic's configuration file and where you do you want them to come from? You can pass in environment variables when you run the container (e.g. with `--env`) and then [access them in configuration](https://torsion.org/borgmatic/docs/how-to/provide-your-passwords/#environment-variable-interpolation).

**Reply by notflip** [reply]
Thanks @witten I want to pass, for example the URL of my healthchecks ping, so if I undestand correctly I can add it to my .env file 
`
PING_URL=https://healthchecks.whatever
`

and then use

```
healthchecks:
  ping_url: ${PING_URL} 
```

**Reply by witten** [reply]
Yeah that sounds right, assuming your `.env` file is used to create your containers (via `docker run` or Docker Compose).
